plugins/woocommerce-admin/client/products/fields/attribute-term-input-field/attribute-term-input-field.tsx

```diff
@@ -0,0 +1,172 @@
+/**
```

Comment:
Hmmm this is actually a tough one from my perspective, as the ProductAttribute returns a list of terms as an array of strings by name (not slugs 😢 ), it would be a bit strange to change the entire field to only use a string array, as there could be duplicate names, and in general it's not specific enough.

An option could be to add an option to pass the product id in and retrieve all the terms by product id ( the `getProductAttributeTerms` api allows for this) and use that as the value instead, but then we have a slight dilemma on what the source of truth is, the `value` prop or the `terms` coming from the data store attached to the specific product id.

My proposal would be to leave this as is, and handle the retrieving of the `ProductAttributeTerm` outside of this field, in a similar manner as what I do now, I only set the `options` prop in the Modal not in the AttributeTerm field.

Retrieving all the product attributes for the edited product should be relatively easy by doing:
```
getProductAttributeTerms({
  atttribute_id: attributeId,
  product: productId
});
``` 
You could trigger that within the edit modal and pass those values straight into the ProductAttributeTermsInputField.
Hope that makes sense? and does my reasoning seem ok in this regard?

Reply:
> the ProductAttribute returns a list of terms as an array of strings by name (not slugs cry ),

By this, do you mean the API? Definitely annoying that it wouldn't include the slugs 🤔

I'm fine with this approach, although it feels mildly painful to throw another request in. That makes sense given our limitations.

Reply:
> By this, do you mean the ProductAttribute API?

I actually mean the Products API, if you look at the **Product - Attributes properties** type here: https://woocommerce.github.io/woocommerce-rest-api-docs/#product-properties you can see it returns a list of options (an array of strings) which is a "list of available term names of the attribute.".

This would mean you did have to trigger a request anyhow to display the items correctly within the typeahead field. If names were unique this wouldn't be as much of a problem and we could probably rely on names only, but given that is not the case, we should definitely retrieve the full attribute term object using the attribute term API endpoint.